Fix for MacOS Shift/Scroll Issue

desagas opened the pull request:
This works because A. minecraft and forge currently do not us xoffset as it relates to scrollCallback(), and B. Because the default  MAC config is to HScroll when pressing shift and scrolling at the same time. This change very simply combines them together, which will complete fix the issue.

- GLFW does in fact see and output both offsets, so this is not a GLFW problem, this is a minecraft problem.

desagas committed:
Fix for the Mac Shift + Scroll = HScroll

This works because A. minecraft and forge currently do not us xoffset as it relates to scrollCallback(), and B. Because the default  MAC config is to HScroll when pressing shift and scrolling at the same time. This change very simply combines them together, which will complete fix the issue.

- GLFW does in fact see and output both offsets, so this is not a GLFW problem, this is a minecraft problem.

desagas committed:
Merge pull request #1 from desagas/desagas-MouseHelperMacOSFix-1

Fix for the Mac Shift + Scroll = HScroll

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/MinecraftForge/MinecraftForge?pullRequest=7590) <br/>All committers have signed the CLA.

Cyborgmas commented:
Yeah you arent supposed to manually edit patch files... you have to fork the repo, run `gradlew setup` in the console then you can go and edit the minecraft source file (in the `forge` project not the `clean` one) and run `gradlew genPatches`. 
The patch file will be generated automatically. Note that it is desired they are as small as possible so you want to not put any imports in them and instead always use the FQN.

desagas commented:
Okay, so, with no import, should I just put the "get os operating system" function in manually, to get passed that?

Cyborgmas commented:
its just instead of doing
`Class.doStuff()` go with `net.package.mine.Class.doStuff()`
the first will require an import the 2nd not

desagas commented:
Okay, so I would want to combine the Util package, and use it for all three references that will be imported in total.

desagas commented:
Gotcha. Thanks

desagas commented:
Hmmm, gradle didn't manager to install the net/minecraft files. I can't quite figure out how to get the to show up.

desagas commented:
Is it normal for the `gradlew setup` to delete a line of code. In the most recent build (38), `public boolean isMiddleDown() {
      return this.middleDown;
   }` has been removed from the MouseHelper code ... so< i guess, temporarily, I added it back so that I could test my code, as ForgeClientHooks can not run without that one line of code.

desagas committed:
Part two, Not Manual, FIXES SHIFT SCROLL MAC

So, I have run this through gradlew genPatches, I have also tested it, and Forge, I can finally say that LSHIFT and SCROLL on mac FINALLY work.

desagas commented:
Something went wrong, I am just going to cancel this, refork, and resubmit.
